File: Agents/jason_agent/src/env/ontologies/HMAS.java
package ontologies;

import org.eclipse.rdf4j.model.IRI;
import org.eclipse.rdf4j.model.ValueFactory;
import org.eclipse.rdf4j.model.impl.SimpleValueFactory;

public class HMAS{

    private static ValueFactory rdf = SimpleValueFactory.getInstance();

    public static IRI Signifier = rdf.createIRI("https://purl.org/hmas/Signifier");

    public static IRI Annotation = rdf.createIRI("https://purl.org/hmas/Annotation");

    public static IRI recommendsAbility = rdf.createIRI("https://purl.org/hmas/recommendsAbility");

    public static IRI signifies = rdf.createIRI("https://purl.org/hmas/signifies");

    public static IRI conveys = rdf.createIRI("https://purl.org/hmas/conveys");

    public static IRI hasGoal = rdf.createIRI("https://purl.org/hmas/hasGoal");

    public static IRI Agent = rdf.createIRI("https://purl.org/hmas/Agent");

    public static IRI hasAbility = rdf.createIRI("https://purl.org/hmas/hasAbility");

    public static IRI hasInteractionPolicy = rdf.createIRI("https://purl.org/hmas/hasInteractionPolicy");

    public static IRI RecurrentPolicy = rdf.createIRI("https://purl.org/hmas/RecurrentPolicy");

    public static IRI hasAnnotationUrl = rdf.createIRI("https://purl.org/hmas/hasAnnotationUrl");

    public static IRI hasCallbackUrl = rdf.createIRI("https://purl.org/hmas/hasCallbackUrl");

    public static IRI MessagePolicy = rdf.createIRI("https://purl.org/hmas/MessagePolicy");

    public static IRI hasMessageUrl = rdf.createIRI("https://purl.org/hmas/hasMessageUrl");

    public static IRI hasId = rdf.createIRI("https://purl.org/hmas/hasId");

    public static IRI registerProfile = rdf.createIRI("https://purl.org/hmas/registerProfile");

    public static IRI queryAnnotations = rdf.createIRI("https://purl.org/hmas/queryAnnotations");

    public static IRI isProfileOf = rdf.createIRI("https://purl.org/hmas/isProfileOf");

    public static IRI Message = rdf.createIRI("https://purl.org/hmas/Message");

    public static IRI hasSender = rdf.createIRI("https://purl.org/hmas/hasSender");

    public static IRI hasReceiver = rdf.createIRI("https://purl.org/hmas/hasReceiver");

}

File: Environment/mcp-server/ontologies/HMAS.py
from rdflib import Namespace

# Base namespace for HMAS terms (note trailing slash so terms append correctly)
HMAS = Namespace("https://purl.org/hmas/")

# Classes / properties
recommendsAbility   = HMAS.recommendsAbility
ActionSpecification = HMAS.ActionSpecification
ActionExecution     = HMAS.ActionExecution
conveys             = HMAS.conveys
signifies           = HMAS.signifies
hasId               = HMAS.hasId
recommendsContext   = HMAS.recommendsContext
ResourceProfile     = HMAS.ResourceProfile
expects             = HMAS.expects
hasForm             = HMAS.hasForm
exposesSignifier    = HMAS.exposesSignifier

Signifier           = HMAS.Signifier
Annotation          = HMAS.Annotation
Agent               = HMAS.Agent
Artifact            = HMAS.Artifact
hasState            = HMAS.hasState
hasProcedure        = HMAS.hasProcedure
hasPrecondition     = HMAS.hasPrecondition
hasOperation        = HMAS.hasOperation
hasPostcondition    = HMAS.hasPostcondition
hasGoal             = HMAS.hasGoal
hasAbility          = HMAS.hasAbility
hasCreator          = HMAS.hasCreator

Generation          = HMAS.Generation
Transformation      = HMAS.Transformation
hasCodeLocation     = HMAS.hasCodeLocation

hasAnnotationPolicy = HMAS.hasAnnotationPolicy
hasInteractionPolicy = HMAS.hasInteractionPolicy
NoKnownAnnotation   = HMAS.NoKnownAnnotation
hasKnownAnnotation  = HMAS.hasKnownAnnotation

RecurrentPolicy = HMAS.RecurrentPolicy
hasRecurrentPolicy = HMAS.hasRecurrentPolicy
hasRepetitionTime = HMAS.hasRepetitionTime
hasCallbackUrl = HMAS.hasCallbackUrl
registerProfile = HMAS.registerProfile
queryAnnotations = HMAS.queryAnnotations


__all__ = [
    "HMAS",
    "recommendsAbility",
    "ActionSpecification",
    "ActionExecution",
    "signifies",
    "recommendsContext",
    "ResourceProfile",
    "expects",
    "hasForm",
    "exposesSignifier",
    "Signifier",
    "Agent",
    "Artifact",
    "hasState",
    "hasProcedure",
    "hasPrecondition",
    "hasOperation",
    "hasPostcondition",
    "hasGoal",
    "hasInteractionPolicy",
    "RecurrentPolicy",
    "registerProfile",
    "queryAnnotations",
]
